### PR TITLE
maybe_error in monad.hpp

### PR DESCRIPTION
**Pull request by tylerjw**

### CODE REVIEW

File: include/rsl/monad.hpp [hunk 1]
```diff
@@ -183,3 +183,25 @@ template <typename T, typename Fn, typename = std::enable_if_t<!rsl::is_optional
...
+ *
+ * @param[in]   The tl::expected<T, E> variables.  All have to use the same
+ * error type.
+ * @tparam      E The error type
+ * @tparam      Args The value types for the tl::expected<T, E> args
+ * @return      The first error found or nothing
+ * @example     maybe_error.cpp
+ */
+template <typename E, typename... Args>
+constexpr std::optional<E> maybe_error(tl::expected<Args, E>... args) {
```
Comment (ChrisThrasher) [inline include/rsl/monad.hpp:198]:
```suggestion
/**
 * @brief Tests if any of the expected args passed in has an error.
 *
 * @param[in]   tl::expected<T, E> parameter pack
 *
 * @tparam E    The error type
 * @tparam Args The value types for the tl::expected<T, E> args
 *
 * @return The first error found or nothing
 */
template <typename E, typename... Args>
[[nodiscard]] constexpr auto maybe_error(tl::expected<Args, E>... args) {
```

File: tests/monad.cpp [hunk 2]
```diff
@@ -228,3 +228,15 @@ TEST_CASE("operator|") {
...
+TEST_CASE("rsl::maybe_error") {
+    SECTION("Error") {
+        auto const exp = tl::expected<int, double>(tl::unexpected(0.1));
+        CHECK(rsl::maybe_error(exp));
+    }
+
+    SECTION("Value") {
+        auto const exp = tl::expected<int, double>(1);
+        CHECK_FALSE(rsl::maybe_error(exp));
+    }
```
Comment (ChrisThrasher) [inline tests/monad.cpp:241]: We need to test with zero parameters and more than 1 parameter since this is a parameter pack. Also test with parameter packs that have different numbers of unexpecteds.

**Review by ChrisThrasher — CHANGES_REQUESTED** [review]

**Comment by ChrisThrasher** [comment]
See #107
